### PR TITLE
Add documentation to address how to contribute to stglib

### PR DESCRIPTION
**Pull request by mmartini-usgs**
Here is a file for the docs that describes how to contribute.

- A requirements file, doc-requirements.yml was added, specific to building docs.  This was provided so that conda can build an environment for making docs
- A make.bat file was added so that docs can be built on windows machines.
- index.rst was updated to include contributing.rst

I have tested the docs build and it works.

### CODE REVIEW

**Comment by dnowacki-usgs** [comment]
Marinna, thank you for this contribution! It looks like there are a few commits not related to the docs in this PR. I think (and correct me if I'm wrong) you may have wanted this PR only to reference commit 65b044a.

That commit looks pretty awesome! Things seem solid to me, except I am not sure that I would suggest folks install the IOOS environment for stglib development. Instead, I would rather have folks install a conda environment using the requirements.yml file in the root of stglib. This is outlined in the [docs](https://stglib.readthedocs.io/en/latest/install.html). What do you think?

**Comment by mmartini-usgs** [comment]
@dnowacki-usgs - yes, IOOS is pretty specific.  @rsignell-usgs: How about getting stglib on conda-forge?  

A good goal to get to is to have a requirements.yml that allows a single install process, rather than install say, IOOS from conda-forge, then some other miscellaneous packages (sqlite is one) at the command line, sometimes using pip.  It is possible to build a .yml file that will mix conda with pip, a little advanced, and doable.

**Comment by mmartini-usgs** [comment]
>  It looks like there are a few commits not related to the docs in this PR. 
Hi Dan - I see one commit including 4 files (https://github.com/dnowacki-usgs/stglib/commit/65b044aeb6e8080ed3a312198efbbb6e832a606e).  4 files would be correct as follows:

- contributing.rst is the main doc(ument)
- index.rst was updated to include a link to contributing.rst
- the "how to contribute" documentation instructs contributors to make a separate environment to build the docs.  doc-requirements.yml is provided for that, there wasn't a yml in the doc directory.  I have tested this and it works.
- make.bat is included to be inclusive of users on windows environments, so they can build the docs and test any additional documentation changes locally. makefile is not helpful for windows.  I have tested this and it runs.

**Comment by dnowacki-usgs** [comment]
Got it, weird, the way GitHub represents the updating of master to the branch is confusing.

For me, and for Appveyor/Travis, the existing stglib requirements.yml files (e.g. [requirements-py37.yml](https://github.com/dnowacki-usgs/stglib/blob/master/requirements-py37.yml)) create an environment with all necessary dependencies, including sqlite, ready for installing stglib following the [docs](https://stglib.readthedocs.io/en/latest/install.html). Are you experiencing otherwise? 

Getting stglib on conda-forge is indeed [a goal](https://github.com/dnowacki-usgs/stglib/issues/50); I just haven't done it yet and am a bit intimidated by the process. If you have seen a good tutorial for doing this I would be interested to read it!

Overall, I'm ready to merge this if we can come to a consensus on the installing part. How does that sound?

**Comment by mmartini-usgs** [comment]
I ended up with the doc building specific environment [doc-requirements.yml](https://github.com/mmartini-usgs/stglib/blob/how_to_contrib/doc/doc-requirements.yml) because 

- following xarray's example, that's what they did 
- the following packages are needed to build the docs:  sphinx-argparse, sphinx_rtd_theme and numpydoc.  Those are not in [requirements-py*.yml](https://github.com/dnowacki-usgs/stglib/blob/master/requirements-py37.yml).  

So if you wanted just the one yml file, you'd have to indicate somewhere that these packages are needed for the doc building.  

There's more than one way to do this, I followed the xarray example.

**Comment by dnowacki-usgs** [comment]
Thanks Marinna. The separate requirements for the docs makes sense. To clarify, I was suggesting that we not have users install the full IOOS environment if we can just have them install the requirements for stglib via the install instructions already on readthedocs. For most users this will soon be moot as stglib is now on conda-forge (though the run scripts do have issues which I am trying to straighten out). 

Basically, if you're ok with my changing the lines outlined in the code review above to point people to the install page on readthedocs, then I'm ready to merge this.

File: doc/contributing.rst [hunk 1]
```diff
@@ -0,0 +1,535 @@
...
+
+Before starting any development, you'll need to create an isolated stglib
+development environment:
+
+- We recommend installing the IOOS conda environment per these instructions\:
+  `Installing the IOOS Environment <http://ioos.github.io/notebooks_demos/other_resources/>`_
+- Make sure your conda is up to date with the command (``conda update conda``)
+- Make sure that you have cloned the repository
+- ``cd`` to the *stglib* source directory (your fork, locally, on your own machine)
+- install *stglib* per `Installation <https://stglib.readthedocs.io/en/latest/install.html>`
```
[the remark this answers is not in the document]
Comment (mmartini-usgs) [reply doc/contributing.rst]: Sure, that's fine.
